src/rust/src/params.rs

```diff
@@ -0,0 +1,1440 @@
+use args::{Args, OutFormat};
+use num_integer::Integer;
+
+use std::fs::File;
+use std::io::{prelude::*, BufReader};
+use std::string::String;
+
+#[cfg(windows)]
+use std::os::windows::io::{AsRawHandle, FromRawHandle};
+use structs::CcxSOptions;
+use time::OffsetDateTime;
+#[cfg(windows)]
+use winapi::um::ioapiset::SetFileCompletionNotificationModes;
+#[cfg(windows)]
+use winapi::um::winbase::FILE_FLAG_OVERLAPPED;
+#[cfg(windows)]
+use winapi::um::winbase::{FILE_SKIP_COMPLETION_PORT_ON_SUCCESS, FILE_SKIP_SET_EVENT_ON_HANDLE};
+#[cfg(windows)]
+use winapi::um::winnt::{FILE_READ_ATTRIBUTES, FILE_SHARE_DELETE, FILE_SHARE_WRITE};
+#[cfg(windows)]
+use winapi::um::winnt::{FILE_SHARE_READ, GENERIC_READ, HANDLE};
+
+#[cfg(windows)]
+unsafe fn set_binary_mode() {
+    let stdin_handle: HANDLE = libc::get_osfhandle(libc::STDIN_FILENO);
+
+    SetFileCompletionNotificationModes(
+        stdin_handle,
+        FILE_SKIP_COMPLETION_PORT_ON_SUCCESS | FILE_SKIP_SET_EVENT_ON_HANDLE,
+    );
+
+    let mut flags: u32 = 0;
+    let handle = libc::get_osfhandle(0);
+    flags |= FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE;
+    let access_mode = GENERIC_READ;
+    if access_mode != FILE_READ_ATTRIBUTES {
+        flags |= FILE_FLAG_OVERLAPPED;
+    }
+    SetFileCompletionNotificationModes(
+        handle as HANDLE,
+        FILE_SKIP_COMPLETION_PORT_ON_SUCCESS | FILE_SKIP_SET_EVENT_ON_HANDLE,
+    );
+}
+
+use crate::args::{self, OutputField};
+use crate::ccx_encoders_helpers::{
+    capitalization_list, profane, CAPITALIZED_BUILTIN, PROFANE_BUILTIN,
+};
+use crate::structs;
+use crate::{
+    activity::activity_report_version,
+    args::{Codec, Ru},
+    enums::CcxDebugMessageTypes,
+    structs::*,
+};
+
+#[cfg(target_os = "windows")]
+const DEFAULT_FONT_PATH: &str = "C:\\\\Windows\\\\Fonts\\\\calibri.ttf";
+#[cfg(target_os = "windows")]
+const DEFAULT_FONT_PATH_ITALICS: &str = "C:\\\\Windows\\\\Fonts\\\\calibrii.ttf";
+
+#[cfg(target_os = "macos")]
+const DEFAULT_FONT_PATH: &str = "/System/Library/Fonts/Helvetica.ttc";
+#[cfg(target_os = "macos")]
+const DEFAULT_FONT_PATH_ITALICS: &str = "/System/Library/Fonts/Helvetica-Oblique.ttf";
+
+#[cfg(not(any(target_os = "windows", target_os = "macos")))]
+const DEFAULT_FONT_PATH: &str = "/usr/share/fonts/truetype/noto/NotoSans-Regular.ttf";
+#[cfg(not(any(target_os = "windows", target_os = "macos")))]
+const DEFAULT_FONT_PATH_ITALICS: &str = "/usr/share/fonts/truetype/noto/NotoSans-Italic.ttf";
+
+static mut FILEBUFFERSIZE: i64 = 1024 * 1024 * 16;
+static mut MPEG_CLOCK_FREQ: i64 = 0;
+static mut USERCOLOR_RGB: String = String::new();
+static mut UTC_REFVALUE: u64 = 0;
+const CCX_DECODER_608_SCREEN_WIDTH: u16 = 32;
+
+pub trait FromStr {
+    #[allow(dead_code)]
+    fn to_ms(&self) -> CcxBoundaryTime;
+}
```

Comment:
Is this trait required? Can you not add a function `to_ms(value: &str) -> CcxBoundaryTime` instead?